io.openems.edge.battery.fenecon.home/src/io/openems/edge/battery/fenecon/home/FeneconHomeBatteryImpl.java

```diff
@@ -294,111 +330,286 @@ protected ModbusProtocol defineModbusProtocol() throws OpenemsException {
+										ElementToChannelConverter.SCALE_FACTOR_MINUS_1), // [Ah]
+								m(generateBcuChannel(i, "_REMAINING_CAPACITY"),
+										new UnsignedWordElement(towerOffset + 21), //
+										ElementToChannelConverter.SCALE_FACTOR_MINUS_1), // [Ah]
+								m(generateBcuChannel(i, "_MAX_CELL_VOLTAGE_LIMIT"),
+										new UnsignedWordElement(towerOffset + 22)), //
+								m(generateBcuChannel(i, "_MIN_CELL_VOLTAGE_LIMIT"),
+										new UnsignedWordElement(towerOffset + 23))));
+			}
+		} catch (OpenemsException e) {
+			this.log.info("Dynamic Channels could not created");
+		}
+
+		this.channelMap = this.createCellVoltAndTempDynamicChannels(numberOfTowers, numberOfModulePerTower);
+		try {
+			// Register is available -> add Registers for current hardware to protocol
+			int offset = ModuleParameters.ADDRESS_OFFSET_FOR_CELL_VOLT_AND_TEMP.getValue();
+			int voltOffset = ModuleParameters.VOLTAGE_ADDRESS_OFFSET.getValue();
+			int voltSensors = ModuleParameters.VOLTAGE_SENSORS_PER_MODULE.getValue();
+			for (int t = 1; t <= numberOfTowers; t++) {
+				String towerString = "TOWER_" + t + "_OFFSET";
+				int towerOffset = ModuleParameters.valueOf(towerString).getValue();
+				for (int i = 1; i < numberOfModulePerTower + 1; i++) {
+					Collection<AbstractModbusElement<?>> elements = new ArrayList<>();
+					for (int j = 0; j < voltSensors; j++) {
+						String key = this.getSingleCellPrefix(j, i, t) + KEY_VOLTAGE;
+						UnsignedWordElement uwe = new UnsignedWordElement(towerOffset + i * offset + voltOffset + j);
+						AbstractModbusElement<?> ame = m(this.channelMap.get(key).channelId(), uwe);
+						elements.add(ame);
+					}
+					this.getModbusProtocol().addTask(new FC3ReadRegistersTask(towerOffset + offset * i + voltOffset,
+							Priority.HIGH, elements.toArray(new AbstractModbusElement<?>[0])));
+				}
+			}
+
+			int tempOffset = ModuleParameters.TEMPERATURE_ADDRESS_OFFSET.getValue();
+			int tempSensors = ModuleParameters.TEMPERATURE_SENSORS_PER_MODULE.getValue();
+			for (int t = 1; t <= numberOfTowers; t++) {
+				String towerString = "TOWER_" + t + "_OFFSET";
+				int towerOffset = ModuleParameters.valueOf(towerString).getValue();
+				for (int i = 1; i < numberOfModulePerTower + 1; i++) {
+					Collection<AbstractModbusElement<?>> elements = new ArrayList<>();
+					for (int j = 0; j < tempSensors; j++) {
+						String key = this.getSingleCellPrefix(j, i, t) + KEY_TEMPERATURE;
+						UnsignedWordElement uwe = new UnsignedWordElement(towerOffset + i * offset + tempOffset + j);
+						AbstractModbusElement<?> ame = m(this.channelMap.get(key).channelId(), uwe);
+						elements.add(ame);
+					}
+					this.getModbusProtocol().addTask(new FC3ReadRegistersTask(towerOffset + offset * i + tempOffset,
+							Priority.HIGH, elements.toArray(new AbstractModbusElement<?>[0])));
+				}
+			}
+		} catch (OpenemsException e) {
+			this.log.info("Cell Temperature and Voltage Dynamic Channels could not created");
```

Comment:
This should never happen. Add `e.printstacktrace()` so that it becomes for visible in the log.